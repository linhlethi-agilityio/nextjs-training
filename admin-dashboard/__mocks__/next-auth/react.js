export const useSession = () => ({
  data: {
    user: {
      token: 'access-token',
      email: 'user@gmail.com',
      id: '1',
      name: 'User',
      role: 'admin',
    },
    expires: Date.now().toString(),
  },
  status: 'authenticated',
  update: async () => null,
});
