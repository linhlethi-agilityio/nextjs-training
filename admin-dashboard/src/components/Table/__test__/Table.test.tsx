// Components
import { render } from '@testing-library/react';

// Components
import { Table, TableColumnType } from '..';

interface Customer {
  idOrder: string;
  customer: string;
  product: string;
  status: string;
}

const columnsTable: TableColumnType<Partial<Customer>>[] = [
  {
    header: 'Id Order',
    accessor: 'idOrder',
  },
  {
    header: 'Product',
    accessor: 'product',
  },
  {
    header: 'Customer',
    accessor: 'customer',
  },
  {
    header: 'Status',
    accessor: 'status',
  },
];

const data = [
  {
    idOrder: '345-645',
    customer: 'Iphone 11 256 GB TAM',
    product: 'Surabaya',
    status: 'Rejected',
  },
  {
    idOrder: '465-674',
    customer: 'Gaming Chair Dragon War',
    product: 'Surabaya',
    status: 'Complete',
  },
];

describe('Table', () => {
  it('should match snapshot for Table', () => {
    const { container } = render(
      <Table isLoading={false} columns={columnsTable} data={data} />,
    );

    expect(container).toMatchSnapshot();
  });
});
