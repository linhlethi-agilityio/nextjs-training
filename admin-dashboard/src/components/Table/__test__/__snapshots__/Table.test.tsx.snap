// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table should match snapshot for Table 1`] = `
<div>
  <table
    class="chakra-table css-1t9zrjp"
  >
    <thead
      class="css-0"
    >
      <tr
        class="css-0"
      >
        <th
          class="css-1otffnp"
        >
          Id Order
        </th>
        <th
          class="css-1otffnp"
        >
          Product
        </th>
        <th
          class="css-1otffnp"
        >
          Customer
        </th>
        <th
          class="css-1otffnp"
        >
          Status
        </th>
      </tr>
    </thead>
    <tbody
      class="css-0"
    >
      <tr
        class="css-0"
      >
        <td
          class="css-zpl621"
        >
          345-645
        </td>
        <td
          class="css-zpl621"
        >
          Surabaya
        </td>
        <td
          class="css-zpl621"
        >
          Iphone 11 256 GB TAM
        </td>
        <td
          class="css-zpl621"
        >
          Rejected
        </td>
      </tr>
      <tr
        class="css-0"
      >
        <td
          class="css-zpl621"
        >
          465-674
        </td>
        <td
          class="css-zpl621"
        >
          Surabaya
        </td>
        <td
          class="css-zpl621"
        >
          Gaming Chair Dragon War
        </td>
        <td
          class="css-zpl621"
        >
          Complete
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
